go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -0,0 +1,509 @@
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	anpapi "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+	anpclientset "sigs.k8s.io/network-policy-api/pkg/client/clientset/versioned"
+	anpinformer "sigs.k8s.io/network-policy-api/pkg/client/informers/externalversions/apis/v1alpha1"
+	anplister "sigs.k8s.io/network-policy-api/pkg/client/listers/apis/v1alpha1"
+)
+
+const (
+	// maxRetries is the number of times a object will be retried before it is dropped out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the
+	// sequence of delays between successive queuings of an object.
+	//
+	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
+	maxRetries = 15
+)
+
+// Controller holds the fields required for ANP controller
+// taken from k8s controller guidelines
+type Controller struct {
+	// name of the controller that starts the ANP controller
+	// (values are default-network-controller, secondary-network-controller etc..)
+	controllerName string
+	sync.Mutex
+	anpClientSet anpclientset.Interface
+
+	// libovsdb northbound client interface
+	nbClient      libovsdbclient.Client
+	eventRecorder record.EventRecorder
+	// An address set factory that creates address sets
+	addressSetFactory addressset.AddressSetFactory
+	// pass in the isPodScheduledinLocalZone util from bnc - used only to determine
+	// what zones the pods are in.
+	// isPodScheduledinLocalZone returns whether the provided pod is in a zone local to the zone controller
+	// So if pod is not scheduled yet it is considered remote. Also if we can't fetch node from kapi and determing the zone,
+	// we consider it remote - this is ok for this controller as this variable is only used to
+	// determine if we need to add pod's port to port group or not - future updates should
+	// take care of reconciling the state of the cluster
+	// TODO(tssurya): Revisit this in future uniformly across the code base if needed.
+	isPodScheduledinLocalZone func(*v1.Pod) bool
+	// store's the name of the zone that this controller belongs to
+	zone string
+
+	// anp name is key -> cloned value of ANP kapi is value
+	anpCache map[string]*adminNetworkPolicyState
+	// If more than one ANP is created at the same priority behaviour is undefined in k8s upsteam.
+	// If more than one ANP is created at the same priority OVNK will only create the first
+	// incoming ANP, rest of them will not be created.
+	// This map tracks anp.Spec.Priority->anp.Name
+	anpPriorityMap map[int32]string
```

Comment:
do you think it may be easier to read/debug if we put plain priorities from the `ANP.Spec` here and evaluate ovn priorities wen creating acls?

Reply:
yeah good point, i tried to do this, but i couldn't because we want to delete the anp from the cache based on priority in `clearAdminNetworkPolicy` since key is priority and we do:
```
	// we can delete the object from the cache now.
	delete(c.anpPriorityMap, anp.priority)
	delete(c.anpCache, anpName)
```
on delete event i don't have access to the object so it was hard to obtain the obj.Spec.Priority BUT I could change anpCache.priority to also be .obj.Spec.Priority but it was nice to keep that cache's priority as the OVN one so that each rule derived from it could be calculated from that.

Reply:
i could store in the ANPPriority and OVNPriority in anpCache; that can help know the ANP priority of the object... WDYT?

Reply:
yeah I thought of changing this field in the `anpCache` too, I would say it won't be worse if we store both

Reply:
sure i am ok here idc which priority is being used, functionally makes 0 difference.
changing this to using spec.priority as per above request.

Reply:
Note that since BANP doesn't have priority and we started sharing the same cache, I will hardcode anpPriority as 0 so that i don't panic in logging places of the shared code..

Reply:
this is done, will show up in next iteration.

Reply:
oops tests need to be updated.. hold before resolving.

Reply:
oh repair also needs to be fixed for this logic..

Reply:
done.